source/tools/VirtualAudioSink.h

```diff
@@ -242,11 +250,27 @@ class VirtualAudioSink : public AudioSinkBase
             // Write in a loop until the callback says we are done.
             IAudioSinkCallback::Result callbackResult
                     = IAudioSinkCallback::Result::Continue;
+            UtilClampAudioBehavior behavior;
+            // init UtilClampBehavior
+            if (Controller.isUtilClampSupported()) {
+                behavior.UtilClampBehavior(mSampleRate, HostTools::getNanoTime(), mCurrentUtilClamp);
```

Comment:
Is here where you say to increase the utilClamp?

Reply:
Yes.

Reply:
Ok, I measured the gap from UtilClamp setting to effective frequency change on dragonboard with Alessio's tool and it is about 5ms. Could be the problem?

Reply:
I measured a much smaller gap on my device.